components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -84,6 +94,28 @@ object TypeConversionHandler {
     }
   }
 
+  def canBeStrictlyConvertedTo(givenType: SingleTypingResult, superclassCandidate: TypedClass): Boolean =
+    handleStrictConversion(givenType, superclassCandidate)
+
+  private def handleStrictConversion(givenType: SingleTypingResult, superclassCandidate: TypedClass) = {
+    handleStringToValueClassConversions(givenType, superclassCandidate) ||
+    handleStrictNumberConversions(givenType.runtimeObjType, superclassCandidate)
+  }
+
+  private def handleStrictNumberConversions(givenClass: TypedClass, superclassCandidate: TypedClass): Boolean = {
+
+    val boxedGivenClass          = ClassUtils.primitiveToWrapper(givenClass.klass)
+    val boxedSuperclassCandidate = ClassUtils.primitiveToWrapper(superclassCandidate.klass)
+    // TODO: This is probably wrong - relying on index of AllNumbers
+    (boxedGivenClass, boxedSuperclassCandidate) match {
+      case (f, t) if ClassUtils.isAssignable(f, t, true) => true
+      case (f, t) if (AllNumbers.contains(f) && AllNumbers.contains(t)) =>
+        AllNumbers.indexOf(f) >= AllNumbers.indexOf(t)
```

Comment:
This logic makes it so that in strict conversion check BigInteger can be assinged to Float or Double, which is incorrect since its like a narrowing conversion. If its corrected you can remove the TODO comment.

Reply:
I havechanged it so we can only assign BigInteger to BigDecimal